management_api_app/service_bus/deployment_status_update.py

```diff
@@ -74,8 +74,26 @@ def create_updated_deployment_document(resource: dict, message: DeploymentStatus
     resource["deployment"]["status"] = message.status
     resource["deployment"]["message"] = message.message
 
+    # although outputs are likely to be relevant when resources are moving to "deployed" status,
+    # lets not limit when we update them and have the resource process make that decision.
+    output_dict = {}
```

Comment:
consider using dict comprehension 
`output_dict = { output.Name : output.Values for output in message.outputs }`

I don't understand the intention though. Seems like you trying to merge dictionaries? You could use update or the merge operator |=

Reply:
Fixed your first point.

On the second, I think the "|" operator was introduced on python 3.9 and we're on 3.8.
I changed the code to use an update instead of creating a new dictionary object. Let me know if this is better.